### ISSUE TITLE
Inconsistent carousels

### ISSUE THREAD

**Issue by LeadSongDog** [internetarchive/openlibrary#496]
Something is amiss at openlibrary/templates/home/index.html
Four of the twelve book carousels do not show edition counts: popular_carousel, returncart_carousel, classics_carousel, and waitlist_carousel. The same four omit the author from the alt text. This could be a big deal for accessibility.

**Comment by mekarpeles** [comment]
@LeadSongDog can you check whether this has been resolved? If so, happy to close. Thank you!

**Comment by LeadSongDog** [comment]
@mekarpeles  Sorry, still not resolved. The carousels "Books we love" "Recently returned" "Romance" "Kids" "Comics" "Thrillers" "Textbooks" "Success" "Technical books" "Classic literature" all show title only. The carousels "Science fiction" "Biographies" and "Mysteries" also show author and edition count.  The titles, authors, and edition count all use the same font, rather than italicized titles, e.g. 
Abraham Lincoln by Carl Sandburg (74 editions)
Terre des hommes by Antoine de Saint-Exupéry (65 editions)

**Comment by LeadSongDog** [comment]
oops, clicked the wrong button...

**Comment by LeadSongDog** [comment]
Hmm. On Safari, the carousels are showing covers with no title, no author, and no edition count. Was this intentional? I hope not, cause it's really dysfunctional

**Comment by LeadSongDog** [comment]
Looking at https://openlibrary.org/search?q=title%3A+%22Barnaby+Rudge%22&mode=everything the first entry's cover alt-text/hovertext reads:
 Cover of: Barnaby Rudge: a tale of the riots of 'eighty
It has no mention of the author or the edition count. Yet in the "Classic Literature" carousel, the same cover is accompanied by hovertext which reads:
 Barnaby Rudge by Charles Dickens
Still with no edition count.

Compare 
https://openlibrary.org/search?q=title%3A+%22The+Time+Machine%22&mode=everything
where the first cover shows hovertext:
Cover of: The Time Machine
and the "Science Fiction" carousel show that work as:
The Time Machine by H. G. Wells (149 editions)

It would seem that it would be simpler and less confusing to reuse the same code for all these cover presentations.

**Comment by mekarpeles** [comment]
👍 on reusing code for these covers -- one consideration is, works and editions have different data available so we'll have to find a way to homogenize it in the case of the edition (where no edition count may be available)

@LeadSongDog may I ask what browser you are you using? On "Books We Love" etc, I am seeing titles of, e.g. "Tom Clancy's Net Force Hidden Agendas *by Tom Clancy*". It's still strange because this should be rendered by the server (so the browser shouldn't matter -- unless, as is the case w/ safari and/or some browsers, the `title` attribute doesn't get rendered the same)

Thank you!

**Comment by mekarpeles** [comment]
Candidate for #603

**Comment by LeadSongDog** [comment]
The hovertext is the same on FF and Chrome when IA access is blocked. Where unblocked Safari/iOS11 the cover image shows and the hovertext isn't seen. Instead a preview of the linked work's page is enabled by several of the iOS accessibility options.

Oddly, the bottom 4 carousels show only works pages while others show only editions. Is this intentional?

**Comment by mekarpeles** [comment]
Yes kind of sort of.

When no cover is set on Open Library, we try to pull the cover from archive.org. We may be able to do something more clever on the backend (like download the cover from archive.org behind the scenes and add it to our coverstore service). Separate issue :)

The first `n` carousels pull their results behind the scenes from archive.org and then fetch their relevant/corresponding works on Open Library to be displayed.

The bottom-most carousels are going to be deprecated. These are generated directly from Open Library subjects and show unavailable and available works iter-mixed.

Ideally, each carousel item would be to a specific Edition: either one which is available or having the shortest waitlist.

**Comment by LeadSongDog** [comment]
Frankly the whole carousel idea seems a distraction to me, particularly if it leads users to "judge a book by it's cover" rather than giving them information on what's inside, yet they've come to dominate the home page.  OL should be *so* much more than just 36 of the popular English books available from IA that the carousels put forth in one dimension each, rather it should be an excellent union catalog: a rich finding tool, to browse adjoining related subjects, or the sequential works of one writer, or the works at a specific time in various places, or different translations of one work. All that is lost with the cover images only.

**Comment by mekarpeles** [comment]
@LeadSongDog, I appreciate this perspective. We aspire to continuously improve as a metadata resource, both for librarians and for the average user. @hornc and @cdrini have been doing amazing work, volunteering contributions almost every day (!) to help us achieve this mission.

Open Library also needs to remain sustainable as a project and provide value to its stakeholders -- our readers and the Internet Archive. One of the reasons we're allowed to continue development is that thousands of readers every day use us as a mechanism for discovery for finding their next book. Book borrows is one way we measure the value we give to our readers and the value we provide Internet Archive. Rather than trying to push books based on an agenda, we try really hard to choose carousels based on the anonymized analytics we receive from site use.

While a bookcover shouldn't be the only information we show to users (we have and are open to all sorts of ideas to display more information users have grown to expect), it is *by far* the number one most compelling thing which users look for, based on our analytics. And given reality has it that we are competing for eyeballs with Amazon, Goodreads (45M monthly users), Overdrive, OCLC Worldcat, and countless other websites, there's a certain degree of wisdom in listening to analytics and user feedback.

I don't say any of this to invalidate any of the feedback you're offering. As a lover of metadata, I'm doing everything I know how (given the resources we have) to help guide us towards a world of richer metadata. We've been working with Wikidata, Betterworldbooks, Onix feed providers, OPDS providers, and have even been building tools to empower our community to help with work/edition. And I'm also happy to invest my person hours, my weekends, reviewing any code contributions you want to offer to make our content more engaging, equitable, discoverable, and accessible to wider audiences.

Rather, I simply hope to communicate that on the other end of the equation, there are tens of thousands of readers on Open Library who are looking for romance novels, self-help books, kids books, and modern thrillers. And that performance and engagement are necessary criteria which determine our future. I know you aren't advocating that we ignore these things, but I think it's easy (and I only say this because I do it all the time) to view Open Library in a purist lends of being the open metdata catalog for the world, without strategizing for the elements which help us survive and hit these milestones. Said with the absolute highest degree of love and respect <3 - mek

**Comment by LeadSongDog** [comment]
I don't for a moment devalue the good work being done, rather I'm hoping to see it being used on the most valuable tasks.  Absolutely, anything that improves reader engagement is welcome, but that should not be the only measure of success. To the extent that the visual interface is permitted to obscure access to the very real riches in the catalog, it becomes just a poorer version of the booksellers' gateways, without their corresponding revenue streams. That`s a relevance death trap for OL.  It will never be slicker than Amazon, Google, and Apple, all of whom have vested interest in selling non-free knowledge.

IA's true competitive advantage lies in the trove of rare content inaccessible to most readers by other means. OL metadata is the best way to let readers find that rare content.  Of course, older works often used title pages that do not render well as thumbnail, especially when in visual competition with the boldly colored modern covers. One tile size certainly "does not fit all" in this use case. If it takes 4x4 carousels instead of 6x6 to solve this, that isn't a big loss to get the win.

Amazon used books, for one example, will often display a different cover than the specific edition being queried for. Similarly its "Look Inside" feature. If one edition of a work has a well-imaged cover and another edition is available for loan but no edition cover image, analytics will argue for showing the best-liked (most clicked) cover image for all editions of a work, then showing "Other editions" on click-through.  All of them use an "Other editions" and/or "You may also like" functionality. Why shouldn't OL do the same sorts of thing?

For the low-graphic user (whether blind, or driving, or perhaps, someday, off-line) the carousels now amount to this:
![carousel2](https://user-images.githubusercontent.com/11435431/32382508-d046c976-c08b-11e7-8876-8dea6154dc36.png)
 
Further, now, the lower of the two shown is about to change to be more like the upper. How is that to be an improvement for these users? Rather, the gateway to the catalog should let them see _how_ one choice is more suited to their interest than another. 

Sorry if I'm rambling a bit here, there's quite a bit to be addressed.
Cheers,
 Dog
----------------------
"The only thing that you absolutely have to know, is the location of the library." - A. Einstein

**Comment by mekarpeles** [comment]
Great points

> If one edition of a work has a well-imaged cover and another edition is available for loan but no edition cover image, analytics will argue for showing the best-liked (most clicked) cover image for all editions of a work

We use a similar heuristic with it's own caveats. Since it's hard to programmatically tell which cover is "best", we currently use the work cover on carousels (and fall back to editions or to IA as necessary). As I see recently returned books without covers, I try to do my part to update them (though this isn't the best leverage of my time) and we're speaking with vendors to find a way to get access to more high quality edition covers in bulk.

Re: size, styling, number of books per carousel and other metadata visually shown; progress has been slowed here mostly for technical reasons of how they are programmed/implemented and time/resource limitations. Related, we're also trying to make pages more mobile responsive -- many of our pages have fragile hard coded absolute positioning in px which conspires against both these agendas and we're relying on an outdated version of jquery which precludes using many drag and drop solutions.

Some of these points are hard to keep current on GitHub but we've been discussing them as a community and figuring out where to spend time on issues during our last 15 weeks or so of 11am PT open library weekly community calls and over our slack channel which is pretty active. We're happy to invite anyone to either. Of course we're not against progress and any pull requests which address these issues we're happy to review. Especially if there's an element about which you feel strongly, make a change and send it our way!


>  "Other editions" and/or "You may also like" functionality. Why shouldn't OL do the same sorts of thing?

We also recently added other books you might like on edition and work pages, and have a development branch which includes a feature to suggest other editions of a book when the page you are on is checked out.

Re: accessibility, you're entirely right. The classics carousel needs to be fixed and made to use consistent title / author messaging.

I'd also like the book title and author, and preferably an edition note, to always show up, especially because many book covers are images of blank/unhelpful contents.

Feel free to create additional issues for each of these things, and especially to tag them with the number #603 so people at the upcoming hackathon can prioritize them.

Thanks again for the great feedback -- in General I'd recommend are slack channel for this type of conversation as it occurs on the fairly daily basis there :)

**Comment by mekarpeles** [comment]
Inconsistent carousels addressed, though we may need better micro/linked-data and better fallback if archive.org blocked (suggesting these get opened as new issues.

**Comment by LeadSongDog** [comment]
Thank you. Cover images are now visible directly from covers.openlibrary.org too vice from archive.org!
